flow/workflows/cdc_flow.go

```diff
@@ -236,25 +222,50 @@ func processTableAdditions(
 				DropFlowStats:         val.DropMirrorStats,
 				SkipDestinationDrop:   val.SkipDestinationDrop,
 			}
-		} else if val.RequestedFlowState == protos.FlowStatus_STATUS_RESYNC {
-			logger.Info("resync requested during table additions")
+		case protos.FlowStatus_STATUS_RESYNC:
+			logger.Info("resync requested", slog.String("operation", op))
 			state.ActiveSignal = model.ResyncSignal
 			// since we are adding to TableMappings, multiple signals can lead to duplicates
 			// we should ContinueAsNew after the first signal in the selector, but just in case
 			cfg.Resync = true
 			cfg.DoInitialSnapshot = true
 			state.DropFlowInput = &protos.DropFlowInput{
 				// to be filled in just before ContinueAsNew
-				FlowJobName:           "",
-				FlowConnectionConfigs: nil,
+				FlowJobName:           cfg.FlowJobName,
+				FlowConnectionConfigs: cfg,
```

Comment:
If this is safe to do, why was the logic otherwise to begin with?

Reply:
still ends up being replaced afterwards, so these values don't matter, but flow name is same either way. tried lifting this here hoping to have more code paths use this function, but they all have their own quirks